matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/permalinks/PermalinkFactory.kt

```diff
@@ -40,11 +41,18 @@ internal class PermalinkFactory @Inject constructor(
         } else MATRIX_TO_URL_BASE + escape(id)
     }
 
-    fun createRoomPermalink(roomId: String): String? {
+    fun createRoomPermalink(roomId: String, via: List<String>? = null): String? {
         return if (roomId.isEmpty()) {
             null
         } else {
-            MATRIX_TO_URL_BASE + escape(roomId) + viaParameterFinder.computeViaParams(userId, roomId)
+            buildString {
+                append(MATRIX_TO_URL_BASE)
+                append(escape(roomId))
+                append(
+                        via?.joinToString(prefix = "?via=", separator = "&via=") { URLEncoder.encode(it, "utf-8") }
```

Comment:
Maybe move this code to `ViaParameterFinder`, since this call to `joinToString` is already there

Reply:
done